packages/lib/reducer.ts

```diff
@@ -1218,7 +1218,28 @@ const reducer = produce((draft: Draft<State> = defaultState, action: any) => {
 					// add it to it.
 					if (!found) {
 						if (isViewingAllNotes || noteIsInFolder(modNote, windowDraft.selectedFolderId)) {
-							newNotes.push(modNote);
+							// When custom sort is active, insert new notes at the top (after to-dos if applicable)
+							// to match Android behavior and improve UX. To-dos are already handled by sortNotes.
+							const isCustomSort = draft.settings['notes.sortOrder.field'] === 'order';
```

Comment:
It may also be necessary to handle the case where "reverse sort order" is enabled.